Update docs and manifests (#251)
sourcegraph/code-intel-extensions: extensions/go/README.md

    - If you're using it with a self-hosted Sourcegraph instance, enter the Sourcegraph instance URL into the Sourcegraph browser extension options menu. Then click the gear icon and enable _Experimental features: Use extensions_.
1. Visit any file on your code host and hover over a token to see a description of the token, a **Go to definition** action (if available), and a **Find references** action.

## Basic code intelligence

This extension comes with built-in code intelligence provided by [search-based heuristics](https://docs.sourcegraph.com/user/code_intelligence/basic_code_intelligence). Because this extension uses text-based heuristics, its definition and reference results may not be precise:

-   "Go to definition" on a token goes to the definition found by [universal-ctags](https://github.com/universal-ctags/ctags), a cross-language parsing suite.
-   "Find references" on a token finds all instances of token (with the same case) in the current repository and other repositories.

These heuristics work well for tokens with unique names, such as `render_to_view` or `TLSConfig`. They do not work well for ambiguous tokens, such as `open` or `file`.

### Large repositories

> Heuristic code intelligence will perform a search query in the commit you are viewing. This may cause performance issues if the commit is not indexed and the repository is large. After a timeout period with no results, an index-only search will be performed. This type of query may return results for a commit other than the one you are currently viewing. The default timeout period is five seconds, but can be lowered by adding the following to your Sourcegraph global settings:

Reviewer: ```suggestion
Basic code intelligence will perform a search query in the commit you are viewing. This may cause performance issues if the commit is not indexed and the repository is large. After a timeout period with no results, an index-only search will be performed. This type of query may return results for a commit other than the one you are currently viewing. The default timeout period is five seconds, but can be lowered by adding the following to your Sourcegraph global settings (units are milliseconds):
```

Since "basic" is the term we defined above, we should probably be consistent here.

Re: Units: I wrote milliseconds but whatever is true here.

Reviewer: (If you accept these changes they'll also need to be propagated back into the template of course)

Author: Updated in all three places.

    ```json
    "basicCodeIntel.unindexedSearchTimeout": 1000
    ````

> For organizations that organize code in a monorepo, it may never be useful to perform an un-indexed search. Index-only search queries can be forced by adding the following to your Sourcgraph global settings.

Reviewer: ```suggestion
For organizations that organize code in a monorepo, it may never be useful to perform an un-indexed search. To force only indexed search queries, add the following to your Sourcgraph global settings.
```

Inspired by https://about.sourcegraph.com/handbook/communication/style_guide#avoid-jargon-and-acronyms, https://about.sourcegraph.com/handbook/communication/style_guide#clarity

Author: Updated in all three places.

    ```json
    "basicCodeIntel.indexOnly": true
    ```

## LSIF

To enable [LSIF support](https://docs.sourcegraph.com/user/code_intelligence/lsif), add these to your Sourcegraph global settings:

```json
  "codeIntel.lsif": true
```

## Language server

> This extension is configured to talk to a language server over WebSockets. On Sourcegraph.com, this extension is already configured. If you are running a

Reviewer: ```suggestion
This extension communicates with a language server over WebSockets. On Sourcegraph.com, this extension is already configured. If you are running a
```

Unless we can configure it to communicate some other way.

Author: Updated in both LSP extensions.

### Using Kubernetes

> If you want to deploy the language server with Kubernetes, your deployment should look like this:

Reviewer: ```suggestion
To deploy the language server with Kubernetes, use a deployment like this:
```

Author: Updated in both LSP extensions.

                  name: cache-ssd
```

With a corresponding service:

```yaml
apiVersion: v1
kind: Service
metadata:
    annotations:
        prometheus.io/port: '6060'
        prometheus.io/scrape: 'true'
    labels:
        app: lang-go
    name: lang-go
    namespace: prod
spec:
    loadBalancerIP: your.static.ip.address
    ports:
        - name: debug
          port: 6060
          targetPort: debug
        - name: lsp
          port: 443
          targetPort: lsp
    selector:
        app: lang-go
    type: LoadBalancer
```

#### TLS

> TLS is optional but recommended for production deployments. It is used if `TLS_KEY` and `TLS_CERT` environment variables are set.

Reviewer: ```suggestion
To enable TLS, set the `TLS_KEY` and `TLS_CERT` environment variables. TLS optional but **strongly recommended** for production deployments.
```

Is `TLS_CERT` a path or an encoded certificate?

Author: Looks like an encoded certificate based on the config in https://github.com/sourcegraph/deploy-sourcegraph-dot-com.

#### Scaling out by increasing the replica count

> You can run multiple instances of the go-langserver and distribute connections between them in Kubernetes by setting `spec.replicas` in the deployment YAML:

Reviewer: ```suggestion
To run multiple instances of the go-langserver and distribute connections between them in Kubernetes, set `spec.replicas` in the deployment YAML:
```

Author: Updated.
